atc/exec/set_pipeline_step.go

```diff
@@ -217,6 +221,25 @@ func (step *SetPipelineStep) run(ctx context.Context, state RunState) error {
 		return nil
 	}
 
+	// conditionally check step
+	if step.policyChecker != nil && step.policyChecker.ShouldCheckHttpMethod("PUT") || step.policyChecker.ShouldCheckAction(atc.SaveConfig) {
+		input := policy.PolicyCheckInput{
+			HttpMethod: "PUT",
```

Comment:
It feels a little awkward to be faking a HTTP request for this. I wonder if we should have this be its own action, similar to `UseImage`?

Reply:
I agree, I mainly kept it for consistency with the current behavior of `SaveConfig`, but happy to refactor into its own action if that's preferred.

Reply:
I also think it's better to define a separate action for this set_pipeline step.

Reply:
Updated to omit `HttpMethod` and use new `RunSetPipeline` action